src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,272 @@
+                                                              Map<String, V> filters) throws IOException {
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+        Field[] fields = entityClass.getDeclaredFields();
+        Validator.validateDatabaseFilters(fields, filters);
+
+        List<T> entities = deserializeEntities(entityClass, Files.readString(databasePath));
+
+        return entities.stream()
+                .filter(entity -> {
+                    for (Map.Entry<String, V> filter : filters.entrySet()) {
+                        String fieldName = filter.getKey();
+                        V expectedValue = filter.getValue();
+                        String getterName = getGetterName(entityClass, fieldName);
+
+                        try {
+                            Method getterMethod = entityClass.getMethod(getterName);
+                            Object actualValue = getterMethod.invoke(entity);
+
+                            if (!expectedValue.equals(actualValue)) {
+                                return false;
+                            }
+                        } catch (IllegalAccessException e) {
+                            logger.error("Unable to access property " + fieldName);
+                            e.printStackTrace();
+                            return false;
+                        } catch (InvocationTargetException | NoSuchMethodException e) {
+                            logger.error("No such method for " + fieldName);
+                            throw new RuntimeException(e);
+                        }
+                    }
+                    return true;
+                })
+                .collect(Collectors.toList());
+    }
+
+    public String getDatabasePath(Class<? extends BaseEntity> entityClass) {
+        var databaseFolder = settings.getDatabasePath();
+        return databaseFolder + File.separator + entityClass.getSimpleName() + "Table" + ".json";
+    }
+
+    private <T extends BaseEntity> void updateEntityFields(T outcomeEntity, T incomeEntity) {
+        Field[] fields = incomeEntity.getClass().getDeclaredFields();
+        for (Field field : fields) {
+            String fieldName = field.getName();
+            String setterName = "set" + Character.toUpperCase(fieldName.charAt(0)) + fieldName.substring(1);
+            String getterName = getGetterName(incomeEntity.getClass(), fieldName);
+
+            try {
+                Method getterMethod = incomeEntity.getClass().getMethod(getterName);
+                Object value = getterMethod.invoke(incomeEntity);
+
+                Method setterMethod = outcomeEntity.getClass().getMethod(setterName, field.getType());
+                setterMethod.invoke(outcomeEntity, value);
+            } catch (IllegalAccessException | NoSuchMethodException e) {
+                logger.error("Unable to access property " + fieldName);
+                e.printStackTrace();
+            } catch (InvocationTargetException e) {
```

Comment:
Please combine all 3 exceptions into one `catch`, I don't see a reason to separately handle them.